app/controllers/application_controller.rb

```diff
@@ -10,7 +10,11 @@ class ApplicationController < ActionController::Base
 
   # What to do if authorization fails
   rescue_from CanCan::AccessDenied do |exception|
-    render :partial => "shared/unauthorized", :locals => {:action => exception.action,:resource=> exception.subject},:status => 403
+    response_data = unauthorized_response_data(exception.action, exception.subject)
+    respond_to do |format|
+      format.html { render :partial => "shared/unauthorized", :locals => response_data, :status => 403 }
```

Comment:
it would be more clear if the the `unauthorized_response_data`, was assigned to a local like: `unauthorized_info`, so this would change the use of `message` below to `unauthorized_info.message`.

Without this, a developer might add new keys to the unauthorized_response_data and not realize this is injecting new (possibly conflicting) local variables in the haml partial.